File: Segundo/Monopoly/civitas_ruby (1)/proyecto_civitas_ruby/lib/sorpresa_ir_carcel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
class SorpresaIrCarcel < Sorpresa
  def initialize(texto,tablero)
    super(texto)
    @tablero=tablero
  end
  
  def aplicarAJugador(actual,todos)
    if(jugadorCorrecto(actual,todos))
      super(actual,todos)
      todos[actual].encarcelar(4) 
    end
  end
  
  def toString
    msg = "Ir self.existe_la_propiedad(ipCarcel"
    return msg
  end
  
  public_class_method :new
end
end

File: Segundo/Monopoly/civitas_java (1)/proyecto_civitas/src/civitas/SorpresaIrCarcel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;
import java.util.ArrayList;
/**
 *
 * @author pc
 */
public class SorpresaIrCarcel extends Sorpresa{
    private Tablero tablero;
    
    SorpresaIrCarcel(Tablero tablero,String texto){
        super(texto);
        this.tablero=tablero;
    }
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual,todos))
        {
            informe(actual,todos);
            todos.get(actual).encarcelar(tablero.getCarcel());
        }
    }
    @Override
    public String toString() {
        return "SorpresaIrCarcel: " + "tablero: " + tablero +super.toString()+ '\n';
    }
}
